spec/index.adoc

```diff
@@ -2399,6 +2401,12 @@ QueryFunc = WasmState -> Trap { cycles_used : Nat; } | Return {
   response : Response;
   cycles_used : Nat;
 }
+CompositeQueryFunc = WasmState -> Trap { cycles_used : Nat; } | Return {
+  new_state : WasmState;
```

Comment:
Should there be a `new_state` here?
We are not committing any changes from composite query calls.

Reply:
The change is not committed, but necessary to process callbacks properly. Otherwise, the callbacks would run on the original canister state and not see changes made before .await.